File: Data Structures & Algorithms/reverse-integer/submission-2.py
class Solution:
    def reverse(self, x: int) -> int:
        MIN = -2147483648
        MAX = 2147483647

        res = 0
        while x != 0:
            t = math.fmod(x, 10)
            x = int(x/10)

            if res > int(MAX/10) or (res == int(MAX/10) and t > math.fmod(MAX, 10)):
                return 0
            
            if res < int(MIN/10) or (res == int(MIN/10) and t < math.fmod(MIN, 10)):
                return 0
            
            res = int(res * 10 + t)
        
        return res

File: Data Structures & Algorithms/number-of-one-bits/submission-2.cpp
class Solution {
public:
    int hammingWeight(uint32_t n) {
        uint32_t mask = 1;
        int res = 0;
        while(n > 0) {
            res += (n & mask);
            n = n >> 1;
        }
        return res;
    }
};
